yt_dlp/extractor/pbs.py

```diff
@@ -297,15 +301,15 @@ class PBSIE(InfoExtractor):
             'playlist_count': 3,
         },
         {
-            'url': 'http://www.pbs.org/wgbh/americanexperience/films/death/player/',
+            'url': 'https://www.pbs.org/wgbh/americanexperience/films/mr-polaroid/',
             'info_dict': {
-                'id': '2276541483',
-                'display_id': 'player',
+                'id': '3100967977',
+                'display_id': '3100967977',
```

Comment:
We can't return the legacy media ID sometimes and the new slug ID other times depending on what URL format is passed; a particular video should only have a single unique ID no matter which of its URL formats were passed to yt-dlp.

And also, If we abandon using the legacy media ID as the video ID, then we still need to return it as one of the `_old_archive_ids` whenever possible or else we break users' download archives.

IMO we should keep the legacy ID as the video ID for now, since that will also be a much simpler fix